### PR TITLE
OP-2028: nctl upgrades in ci

### PR DESCRIPTION
**Pull request by TomVasile**
Adds nctl upgrade scenario 1 to bors branches.
Adds nctl ci upgrade wrapper.
Adds functionality and fixes necessary for running in ci to existing nctl code.

### CODE REVIEW

**Comment by TomVasile** [comment]
bors try

**Comment by TomVasile** [comment]
adding aws cli for this: https://github.com/CasperLabs/buildenv/pull/51

**Comment by TomVasile** [comment]
bors try

**Comment by TomVasile** [comment]
bors try

**Comment by TomVasile** [comment]
bors try

**Comment by TomVasile** [comment]
bors try

File: utils/nctl/sh/utils/queries.sh [hunk 1]
```diff
@@ -89,6 +89,17 @@ function get_chain_latest_block_hash()
         | sed -e 's/^"//' -e 's/"$//'
 }
 
+function get_chain_first_block_hash()
```
Comment (Fraser999) [inline utils/nctl/sh/utils/queries.sh:92]: I guess we don't actually need this function after all?

**Review by Fraser999 — APPROVED** [review]
One minor non-blocking point.  Otherwise LGTM.

File: utils/nctl/sh/utils/queries.sh (hunk 1 above)
Comment (TomVasile) [reply utils/nctl/sh/utils/queries.sh:92]: I left it in, I may need it for something else down the line.

**Comment by TomVasile** [comment]
bors try

**Comment by TomVasile** [comment]
bors try

**Pull request merged**